google/cloud/pubsub/internal/publisher_tracing_connection_test.cc

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+#include "google/cloud/pubsub/internal/publisher_tracing_connection.h"
+#include "google/cloud/pubsub/mocks/mock_publisher_connection.h"
+#include "google/cloud/pubsub/publisher_connection.h"
+#include "google/cloud/internal/opentelemetry.h"
+#include "google/cloud/opentelemetry_options.h"
+#include "google/cloud/testing_util/opentelemetry_matchers.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+namespace {
+
+using ::google::cloud::pubsub::PublisherConnection;
+using ::google::cloud::pubsub_internal::MakePublisherTracingConnection;
+using ::google::cloud::pubsub_mocks::MockPublisherConnection;
+using ::google::cloud::testing_util::InstallSpanCatcher;
+using ::google::cloud::testing_util::OTelAttribute;
+using ::google::cloud::testing_util::SpanCatcher;
+using ::google::cloud::testing_util::SpanHasAttributes;
+using ::google::cloud::testing_util::SpanHasInstrumentationScope;
+using ::google::cloud::testing_util::SpanKindIsClient;
+using ::google::cloud::testing_util::SpanNamed;
+using ::google::cloud::testing_util::SpanWithStatus;
+using ::google::cloud::testing_util::ThereIsAnActiveSpan;
+
+class PublisherTracingConnectionTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    span_catcher_ = InstallSpanCatcher();
+    mock_ = std::make_shared<MockPublisherConnection>();
+  }
+
+  std::shared_ptr<SpanCatcher> span_catcher_;
+  std::shared_ptr<MockPublisherConnection> mock_;
+};
```

Comment:
I recommend you simply create `span_catcher` and `mock` in each test function.  These fixtures that just initialize a few variables are more trouble than anything else.  In a couple of years you will be reading a function in the middle of this file and wonder: "What is `mock_`?"

Reply:
Makes sense to me. Done.